js/plugins/google-cloud/src/gcpOpenTelemetry.ts

```diff
@@ -218,24 +207,75 @@ export class GcpOpenTelemetry {
  * helpful information about how to set up metrics/telemetry in GCP.
  */
 class MetricExporterWrapper extends MetricExporter {
+  private promise = new Promise<void>((resolve) => resolve());
+
   constructor(
     options?: ExporterOptions,
     private errorHandler?: (error: Error) => void
   ) {
     super(options);
   }
 
-  export(
+  async export(
     metrics: ResourceMetrics,
     resultCallback: (result: ExportResult) => void
-  ): void {
-    super.export(metrics, (result) => {
-      if (this.errorHandler && result.error) {
-        this.errorHandler(result.error);
-      }
-      resultCallback(result);
+  ): Promise<void> {
+    await this.promise;
+    this.modifyStartTimes(metrics);
+    this.promise = new Promise<void>((resolve) => {
+      super.export(metrics, (result) => {
+        try {
+          if (this.errorHandler && result.error) {
+            this.errorHandler(result.error);
+          }
+          resultCallback(result);
+        } finally {
+          resolve();
+        }
+      });
+    });
+  }
+
+  selectAggregation(instrumentType: InstrumentType) {
+    if (instrumentType === InstrumentType.HISTOGRAM) {
+      return new ExponentialHistogramAggregation();
+    }
+    return new DefaultAggregation();
+  }
+
+  selectAggregationTemporality(instrumentType: InstrumentType) {
+    return AggregationTemporality.DELTA;
+  }
+
+  /**
+   * Modify the start times of each data point to ensure no
+   * overlap with previous exports.
+   *
+   * Cloud metrics do not support delta metrics for custom metrics
+   * and will convert any DELTA aggregations to CUMULATIVE ones on
+   * export. There is implicit overlap in the start/end times that
+   * the Metric reader is sending -- the end_time of the previous
+   * export will become the start_time of the current export. This
+   * method adds a thousandth of a second to ensure discrete export
+   * timeframes.
+   */
```

Comment:
This comment could benefit from explaining "why" we need to avoid overlapping intervals (i.e. last metric wins for that interval).

Reply:
Added an additional statement about why the overlapping times are problematic.